geowave-analytics/src/main/java/mil/nga/giat/geowave/analytics/kmeans/mapreduce/runners/KMeansDistortionJobRunner.java

```diff
@@ -0,0 +1,232 @@
+import org.apache.accumulo.core.client.ClientConfiguration;
+/*end[ACCUMULO_1.5.1]*/
+//@formatter:on
+import org.apache.accumulo.core.client.mapreduce.AccumuloOutputFormat;
+import org.apache.accumulo.core.client.security.tokens.AuthenticationToken;
+import org.apache.accumulo.core.client.security.tokens.PasswordToken;
+import org.apache.accumulo.core.data.Mutation;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.conf.Configured;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.io.Text;
+import org.apache.hadoop.mapreduce.Job;
+import org.apache.hadoop.mapreduce.lib.input.FileInputFormat;
+import org.apache.hadoop.mapreduce.lib.input.SequenceFileInputFormat;
+import org.apache.hadoop.util.Tool;
+import org.apache.hadoop.util.ToolRunner;
+
+
+/**
+ * 
+ * Calculate the distortation.
+ * 
+ * See Catherine A. Sugar and Gareth M. James (2003).
+ * "Finding the number of clusters in a data set: An information theoretic approach"
+ * Journal of the American Statistical Association 98 (January): 750–763
+ * 
+ * 
+ */
+public class KMeansDistortionJobRunner extends
+		Configured implements
+		Tool,
+		MapReduceJobRunner
+{
+	private int reducerCount = 8;
+	private int k = 1;
+	private Path inputHDFSPath;
+	private String distortationTableName;
+
+	public KMeansDistortionJobRunner() {
+
+	}
+
+	public void setReducerCount(
+			final int reducerCount ) {
+		this.reducerCount = reducerCount;
+	}
+
+	public void setCentroidsCount(
+			final int k ) {
+		this.k = k;
+	}
+
+	public void setDistortationTableName(
+			final String distortationTableName ) {
+		this.distortationTableName = distortationTableName;
+	}
+
+	public void setInputHDFSPath(
+			final Path inputHDFSPath ) {
+		this.inputHDFSPath = inputHDFSPath;
+	}
+
+	@Override
+	public int run(
+			final String[] args )
+			throws Exception {
+		final Configuration conf = super.getConf();
+		final Job job = new Job(
+				conf);
+		final String zookeeper = args[0];
+		final String instance = args[1];
+		final String user = args[2];
+		final String password = args[3];
+		final String namespace = args[4];
+
+		job.setJarByClass(this.getClass());
+
+		job.setJobName("GeoWave K-Means Distortion (" + namespace + ")");
+		job.setInputFormatClass(SequenceFileInputFormat.class);
+		job.setMapperClass(KMeansDistortionMapReduce.KMeansDistortionMapper.class);
+		job.setMapOutputKeyClass(Text.class);
+		job.setMapOutputValueClass(CountofDoubleWritable.class);
+		job.setReducerClass(KMeansDistortionMapReduce.KMeansDistortionReduce.class);
+		job.setCombinerClass(KMeansDistortionMapReduce.KMeansDistorationCombiner.class);
+		job.setOutputKeyClass(Text.class);
+		job.setOutputValueClass(Mutation.class);
+		job.setOutputFormatClass(AccumuloOutputFormat.class);
+		job.setNumReduceTasks(reducerCount);
+		// extends wait time to one hour (default: 600 seconds)
+		final long milliSeconds = 1000 * 60 * 60;
+		job.getConfiguration().setLong(
+				"mapred.task.timeout",
```

Comment:
Not sure why an hour was chosen when the code was first written.  I cannot see any reason why there would be a time out.  As long as the reducer is consuming from the value iterator, the task manager assumes progress and updates counters.   I will drop it down to 15 minutes.  My experience, this type of timeout only occurs if there exists a long computation after iterating through all reducer values or the output format has long blocks of wait time.  This code does not fit these cases.